src/bap_mc/bap_mc.ml

```diff
@@ -103,30 +110,41 @@ let print_bil lift bil_formats mem insn =
   List.iter bil_formats ~f:(fun fmt ->
       printf "%s@." (string_of_bil fmt (bil insn)))
 
-let make_print lift insn_fmt bil_fmt show_kinds mem insn =
-  print_insn insn_fmt insn;
+let make_print lift insn_fmt bil_fmt show_kinds oplen_bases mem insn =
+  print_insn_oplen oplen_bases mem;
+  print_insn insn_fmt insn; 
   print_bil lift bil_fmt mem insn;
   if show_kinds then print_kinds insn
 
-let step print state mem insn _ =
-  print mem insn;
+
+let check disasm_len current_total =
+  if disasm_len=0 then true
+  else current_total < disasm_len
```

Comment:
this is usually written as `disasm_len = 0 || current_total < disasm_len`, but of course using `0` as a sentinel is a bad style